### PR TITLE
Add chain invocation support for 'create variable' code action

### LINKED ISSUE #18326: Code action, "Create Variable" does not work for chain invocations
**Description:**
Code action, "Create Variable" does not work for chain invocations due to https://github.com/ballerina-platform/ballerina-lang/issues/18278

**Steps to reproduce:**
Try to hover over ` lorry.get_color().print(10);` to get code action.

```ballerina

# A Lorry
public type Lorry object {
    # Get colour
    # + a - input
    # + return - a string
    public function get_color(int a) returns Color{
        return new Color();
    }
};

# A Car
public type Color object {
    # Print colour
    # + b - input
    # + return - a string
    public function print(int b) returns string{
        return "red";
    }
};

public function main(string... args) {
    Lorry lorry = new Lorry();
    lorry.get_color().print(10);
}
```
**Affected Versions:**
1.0.0-alpha3

**Related Issues (optional):**
https://github.com/ballerina-platform/ballerina-lang/issues/18278

### PR DESCRIPTION
**Pull request by rasika**
Fixes #18326

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/VariableAssignmentCodeAction.java [hunk 1]
```diff
@@ -223,16 +223,44 @@ private static String getDiagnosedContent(Diagnostic diagnostic, LSContext conte
...
+            }
+            if (tailChar == '(') {
+                pendingLParenthesis--;
+            } else if (tailChar == ')') {
+                pendingLParenthesis++;
+            }
+            content = content.substring(0, content.length() - 1);
+            count++;
+        }
+
```
Comment (NipunaRanasinghe) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/VariableAssignmentCodeAction.java]: Shall we update the logic to handle(ignore) characters inside string literals? :)

Comment (rasika) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/VariableAssignmentCodeAction.java]: Yes 👍 , Need to ignore chars inside strings for the calculation.

Comment (NipunaRanasinghe) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/VariableAssignmentCodeAction.java]: @rasika Maybe it'd be better to add a test case including string literals, if possible :)

Comment (rasika) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/VariableAssignmentCodeAction.java]: Yes, added a test case as well.

**Pull request merged**